redbox-core/redbox/loader/loaders.py

```diff
@@ -18,15 +28,198 @@
+        Keys in the ignore list will be excluded from the result.
+        """
+        if ignore is None:
+            ignore = []
+
+        combined = {}
+
+        ignore_set = set(ignore)
+
+        for key in set(x) | set(y):
+            if key in ignore_set:
+                continue
+
+            if key in x and key in y:
+                if isinstance(x[key], list) or isinstance(y[key], list):
+                    combined[key] = list(
+                        set(x[key] + (y[key] if isinstance(y[key], list) else [y[key]]))
+                    )
+                else:
+                    combined[key] = [x[key], y[key]]
+            elif key in x:
+                combined[key] = x[key]
+            else:
+                combined[key] = y[key]
+
+        return combined
+
+    def _get_file_bytes(self, s3_client: S3Client, file_name: str) -> BytesIO:
+        return s3_client.get_object(Bucket=self.env.bucket_name, Key=file_name)[
+            "Body"
+        ].read()
+
+    def extract_metadata(self):
+        """
+        Extract metadata from first 1_000 tokens
+        """
+        file_bytes = self._get_file_bytes(
+            s3_client=self.s3_client, file_name=self.file_name
+        )
+        url = f"http://{self.env.unstructured_host}:8000/general/v0/general"
+        files = {
+            "files": (self.file_name, file_bytes),
+        }
+        response = requests.post(
+            url,
+            files=files,
+            data={
+                "strategy": "fast",
+                "chunking_strategy": "by_title",
+                "max_characters": self.env.worker_ingest_max_chunk_size,
+                "combine_under_n_chars": self.env.worker_ingest_min_chunk_size,
+                "overlap": 0,
+                "overlap_all": True,
+            },
+        )
+
+        if response.status_code != 200:
+            raise ValueError(response.text)
+
+        elements = response.json()
+
+        if not elements:
+            # if we can't extract metadata, return empty metadata
+            return {"name": "", "description": "", "keywords": [""]}
+
+        else:
+            # Get first 1k tokens of processed document
+            first_n = self.get_first_n_tokens(elements, 1_000)
+
+            # Get whatever metadata we can from processed document
+            doc_metadata = self.get_doc_metadata(chunks=elements, n=3, ignore=None)
+
+            # Generate new metadata
+            self.metadata = self.create_file_metadata(first_n, doc_metadata)
+            if not self.metadata:
+                raise ValueError("LLM failed to extract metadata for this file")
+
+    def create_file_metadata(
+        self, page_content: str, metadata: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Uses a sample of the document and any extracted metadata to generate further metadata."""
+        metadata_chain = (
+            ChatPromptTemplate.from_messages(
```

Comment:
I think this prompt should sit in an env var

Reply:
Do you mean to add in `setting.py`?

Reply:
yes please @saisakul